packages/cli/src/build/config.ts

```diff
@@ -125,26 +130,27 @@ export function getRollupConfigs(pkg: Package, aliases: Aliases) {
     ],
   });
 
-  configs.push({
-    config: getRollupConfig(
-      pkg,
-      pkg.entrypoints,
-      aliases,
-      "node-prod",
-      () => {}
-    ),
-    outputs: [
-      {
-        format: "cjs",
-        entryFileNames: "[name].cjs.prod.js",
-        chunkFileNames: "dist/[name]-[hash].cjs.prod.js",
-        dir: pkg.directory,
-        exports: "named",
-        interop,
-        plugins: cjsPlugins,
-      },
-    ],
-  });
+  if (!hasModuleType)
+    configs.push({
```

Comment:
Those files were also acting as optimizations for node since reading from `process.env` is slow. In ESM-only world, we should probably leverage development/production conditions in `package.json#exports` to apply the same optimization. I know that @mitchellhamilton had some concerns related to that though

Reply:
Good with adding this in. @mitchellhamilton any thoughts?

Reply:
I think it should be done as a separate change to this. The big thing IMO is that the syntax for it shouldn't be `process.env.NODE_ENV` since you could have `process.env.NODE_ENV === 'production'` but not have the `production` condition set.